### PR TITLE
✨ [Amp story] Add `amp-story-page-outlink` component

### LINKED ISSUE #33955: [Page attachments] Add component for `amp-story-page-outlink`
`amp-story-page-attachment` covers two different UX patterns (inline and outlink) and isn't semantic.
A new component name for outlinks will help communicate function.

The new name will be: `amp-story-page-outlink`

`amp-story-page-outlink` will point to the code in `amp-story-page-attachment` for backwards compatibility.

To do this:
1. Register element in `amp-story.js`
2. Create validation rules (can copy paste most validation from `amp-story-page-attachment`)

@ampproject/wg-stories

### PR DESCRIPTION
**Pull request by processprocess**
Context / Fixes #33955

- Registers component pointing to `amp-story-page-attachment.js`
- Test for rendering through `amp-story-page-attachment.js`
- Validation
- Updates html template
- Updates page-advancement
- Updates check for render in `amp-story-page`

### CODE REVIEW

File: extensions/amp-story/1.0/amp-story-page.js [hunk 1]
```diff
@@ -1737,9 +1737,10 @@ export class AmpStoryPage extends AMP.BaseElement {
    * @private
    */
   renderOpenAttachmentUI_() {
-    const attachmentEl = this.element.querySelector(
-      'amp-story-page-attachment'
-    );
+    // AttachmentEl can be either amp-story-page-attachment or amp-story-page-outlink
+    const attachmentEl = this.element.querySelectorAll(
+      'amp-story-page-attachment, amp-story-page-outlink'
+    )[0];
```
Comment (gmajoulet) [inline extensions/amp-story/1.0/amp-story-page.js:1743]: You can actually keep the `querySelector()` and remove the `[0]`, that'll do the same thing

File: extensions/amp-story/validator-amp-story.protoascii [hunk 2]
```diff
@@ -921,6 +921,51 @@ tags: {  # <amp-story-page-attachment> with no href
...
+  attrs: {
+    name: "href"
+    mandatory: true
+    value_url: {
+      protocol: "http"
+      protocol: "https"
+    }
+  }
+  attrs: {
+    name: "cta-image"
```
Comment (gmajoulet) [inline extensions/amp-story/validator-amp-story.protoascii:958]: Aren't the images only for the inline attachment?

Comment (processprocess) [reply extensions/amp-story/validator-amp-story.protoascii:958]: It's for both. 
In this case it's used for a custom preview icon:
Screen Shot 2021-05-03 at 1 01 13 PM

**Review by gmajoulet — APPROVED** [review]

**Comment by processprocess** [comment]
Pinging @twifkak for OWNERS :)

File: examples/amp-story/attachment.html [hunk 3]
```diff
@@ -281,14 +281,14 @@
...
-        <amp-story-page-attachment
+        <amp-story-page-outlink
           title="Link Description"
           href="https://www.google.com"
           theme="custom"
           cta-accent-color="navy"
           cta-accent-element="background"
           layout="nodisplay">
-        </amp-story-page-attachment>
+        </aamp-story-page-outlink>
```
Comment (twifkak) [inline examples/amp-story/attachment.html:291]: typo

**Review by honeybadgerdontcare — APPROVED** [review]

File: extensions/amp-story/validator-amp-story.protoascii [hunk 4]
```diff
@@ -933,6 +933,51 @@ tags: {  # <amp-story-page-attachment> with no href
...
+    value: "nodisplay"
+  }
+  attrs: {
+    name: "theme"
+    value: "custom"
+    value: "dark"
+    value: "light"
+  }
+  attrs: {
+    name: "cta-accent-color"
```
Comment (twifkak) [inline extensions/amp-story/validator-amp-story.protoascii:954]: Slight preference to place attrs in alphabetical order.

File: extensions/amp-story/validator-amp-story.protoascii [hunk 5]
```diff
@@ -933,6 +933,51 @@ tags: {  # <amp-story-page-attachment> with no href
...
+    mandatory: true
+    value_url: {
+      protocol: "http"
+      protocol: "https"
+    }
+  }
+  attrs: {
+    name: "cta-image"
+    value_url: {
+      # protocol: "http"  # TODO(honeybaderdontcare): readd once transformers in prod
```
Comment (twifkak) [inline extensions/amp-story/validator-amp-story.protoascii:972]: @honeybadgerdontcare Do we have a tracking bug for this? (Ditto 2 lines down.)

**Review by twifkak — APPROVED** [review]
validator changes LGTM; just a couple nits

File: extensions/amp-story/validator-amp-story.protoascii (hunk 5 above)
Comment (honeybadgerdontcare) [reply extensions/amp-story/validator-amp-story.protoascii:972]: This was completed in cl/361937561, so the comment is outdated and can be removed from -attachement. It has not been done for -outlink.

Comment (honeybadgerdontcare) [reply extensions/amp-story/validator-amp-story.protoascii:972]: For amp-story-page-outlink this is being done in cl/372160779.

**Comment by processprocess** [comment]
> It seems `amp-story-page-outlink` wasn't added to the list of `child_tag_name_oneof` tags in `amp-story-page`.
> 
> Isn't this change needed to allow the component to be used in a story?

Hi Weston, thanks for pointing this out. This will be added as part of [#34532](https://github.com/ampproject/amphtml/pull/34532)

**Pull request merged**